device_services/generic/power_regulator.md

```diff
@@ -0,0 +1,110 @@
+# Power Regulator Service
+
+Power regulator service acts as a controller responsible for turning the relay on and off for a given part of the duty cycle.
+
+## Service name
+
+`power_regulator`
+
+## Interfaces
+
+| Type | Interface            | Value type | Description                    |
+|------|----------------------|------------|--------------------------------|
+| in   | cmd.lvl.set          | int        | Sets the level value.          |                                                                               
+| in   | cmd.lvl.get_report   | null       | Requests the level value.      |
+| out  | evt.lvl.report       | int        | Reports the level value.       |
+| -    |                      |            |                                |
+| in   | cmd.cycle.set        | int        | Sets the duty cycle value.     |                                                                               
+| in   | cmd.cycle.get_report | null       | Requests the duty cycle value. |
+| out  | evt.cycle.report     | int        | Reports the duty cycle value.  |
+
+## Service properties
+
+| Name         | Type   | Example                            | Description                                                                               |
+|--------------|--------|------------------------------------|-------------------------------------------------------------------------------------------|
+| `max_lvl`    | int    | `100`                              | A maximum supported level value.                                                          |
+| `min_lvl`    | int    | `5`                                | A minimum supported level value.                                                          |
+| `lvl_step`   | int    | `5`                                | A step of level.                                                                          |
+| `sup_cycles` | object | `[{"min":60,"max":1800,"step":1}]` | Supported duty cycles settings in form of array of [`cycle_range`](#definitions) objects. |
```

Comment:
What if sup_cycles cannot be written in linear equation? Currently in new Namron/FH th-reg, cycles has 30min steps, bur Steiner wrote, that it will be changed. The arrays with supported values, won't be better?

Reply:
@Brzezoo 

Read a note below it. What was in Steinar doc previously could be provided as:

[{"min":1800,"max":1800,"step":1},{"min":3600,"max":3600,"step":1},...]

But we will have more power regulators. I see no reason why they would have enum style duty cycle always. This notation is powerful enough to allow any settings to be possible in form of a slider.

Reply:
OK. ACK.